src/backend/services/chat.py

```diff
@@ -368,6 +371,63 @@ def update_conversation_after_turn(
     conversation_crud.update_conversation(session, conversation, new_conversation)
 
 
+def generate_chat_response(
+    session: DBSessionDep,
+    model_deployment_stream: Generator[StreamedChatResponse, None, None],
+    response_message: Message,
+    conversation_id: str,
+    user_id: str,
+    should_store: bool = True,
+    **kwargs: Any,
+) -> NonStreamedChatResponse:
+    """
+    Generate chat response from model deployment non streaming response.
+
+    Args:
+        session (DBSessionDep): Database session.
+        model_deployment_stream (Generator[StreamResponse, None, None]): Model deployment stream.
+        response_message (Message): Response message object.
+        conversation_id (str): Conversation ID.
+        user_id (str): User ID.
+        should_store (bool): Whether to store the conversation in the database.
+        **kwargs (Any): Additional keyword arguments.
+
+    Yields:
+        bytes: Byte representation of chat response event.
+    """
+    stream = generate_chat_stream(
+        session,
+        model_deployment_stream,
+        response_message,
+        conversation_id,
+        user_id,
+        should_store,
+        **kwargs,
+    )
+
+    non_streamed_chat_response = None
+    for event in stream:
```

Comment:
This is because non-streaming API doesn't support multistep, right? Do we plan to support non-streaming multistep in API later? If so, maybe worth adding comment/todo here about that.

Reply:
It actually does! Instead of implementing multistep for streaming and nonstreaming, I added only streaming multistep, and the nonstreaming endpoint simply returns the last event without any of the intermediate steps.

Reply:
Will add a comment to clarify